internal/controller/registry_scan_runner.go

```diff
@@ -0,0 +1,179 @@
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/rancher/sbombastic/api/v1alpha1"
+)
+
+const scanInterval = 1 * time.Minute
+
+// RegistryScanRunner handles periodic scanning of registries based on their scan intervals.
+type RegistryScanRunner struct {
+	client.Client
+}
+
+// Start implements the Runnable interface.
+func (r *RegistryScanRunner) Start(ctx context.Context) error {
+	log := log.FromContext(ctx)
+	log.Info("Starting registry scan runner")
+
+	ticker := time.NewTicker(scanInterval)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			log.Info("Stopping registry scan runner")
+
+			return nil
+		case <-ticker.C:
+			if err := r.scanRegistries(ctx); err != nil {
+				log.Error(err, "Failed to scan registries")
+			}
+		}
+	}
+}
+
+// scanRegistries checks all registries and creates ScanJobs for those that need scanning.
+func (r *RegistryScanRunner) scanRegistries(ctx context.Context) error {
+	log := log.FromContext(ctx)
+
+	var registries v1alpha1.RegistryList
+	if err := r.List(ctx, &registries); err != nil {
+		return fmt.Errorf("failed to list registries: %w", err)
+	}
+
+	log.V(1).Info("Checking registries for scanning", "count", len(registries.Items))
+
+	for _, registry := range registries.Items {
+		if err := r.checkRegistryForScan(ctx, &registry); err != nil {
+			log.Error(err, "Failed to check registry for scan", "registry", registry.Name, "namespace", registry.Namespace)
+
+			continue
+		}
+	}
+
+	return nil
+}
+
+// checkRegistryForScan determines if a registry needs scanning and creates a ScanJob if needed.
+func (r *RegistryScanRunner) checkRegistryForScan(ctx context.Context, registry *v1alpha1.Registry) error {
+	log := log.FromContext(ctx)
+
+	if registry.Spec.ScanInterval.Duration == 0 {
+		log.V(2).Info("Skipping registry with disabled scan interval", "registry", registry.Name)
+
+		return nil
+	}
+
+	lastScanJob, err := r.getLastScanJob(ctx, registry)
+	if err != nil {
+		// If no ScanJob exists, create the initial one
+		if apierrors.IsNotFound(err) {
+			if err = r.createScanJob(ctx, registry); err != nil {
+				return fmt.Errorf("failed to create initial scan job for registry %s: %w", registry.Name, err)
+			}
+			log.Info("Created initial scan job for registry", "registry", registry.Name, "namespace", registry.Namespace)
+
+			return nil
+		}
+
+		return fmt.Errorf("failed to get last scan job for registry %s: %w", registry.Name, err)
+	}
+
+	if !lastScanJob.IsComplete() && !lastScanJob.IsFailed() {
+		log.V(1).Info("Registry has a running ScanJob, skipping.", "registry", registry.Name, "scanJob", lastScanJob)
+
+		return nil
+	}
+
+	if lastScanJob.Status.CompletionTime != nil {
+		timeSinceLastScan := time.Since(lastScanJob.Status.CompletionTime.Time)
+		if timeSinceLastScan < registry.Spec.ScanInterval.Duration {
+			log.V(2).Info("Registry doesn't need scanning yet", "registry", registry.Name, "timeSinceLastScan", timeSinceLastScan)
```

Comment:
Nit: should we raise the log level of this message to keep it "hidden" with the default configuration? I wonder if we're not going to pollute the logs